### ISSUE TITLE
[feature request] Support dune build when bisecting

### ISSUE THREAD

**Issue by ejgallego** [JasonGross/coq-bisect#1]
Hi, after having a look to the script, it seems to me that it could benefit from large speedups by detecting if dune is supported in Coq [I can give an exact commit] and the profiting from Dune's incremental builds which work correctly without having to do any `clean` and moreover they build the required set of deps much faster.

I could provide a safe commit for which dune works fine [last 18 months should be fine] , as indeed if the commit being tested is older clean + make has to be the fallback.

The general setup for bisecting with dune is pretty easy, for a file foo just do:
```
dune build $foo_deps.vo
dune exec -- dev/shim/coqc-prelude $foo.v
```
this will exactly build the required files to correctly compile `foo.vo`

Next dune version will include a `dune coqc` command that combines the 2 and automatically runs `coqdep`, it will work with older Coq, it only needs a newer dune.

**Comment by JasonGross** [comment]
Yeah, this would be pretty great.  I should plausibly also switch to using the Coq docker image so I can switch between different versions of OCaml as needed.  Do you know the magic invocation to tell if the current commit has a given one as it's ancestor?

**Comment by ejgallego** [comment]
> Do you know the magic invocation to tell if the current commit has a given one as it's ancestor?

`git branch --contains` should do the trick, right?

A good commit to set as cut-off point is IMO https://github.com/coq/coq/commit/5db591257070734439dd5550995d6d3f497256c0 

I tested it and branches containing that commit should indeed support fully incremental calling of `dune exec -- dev/shim/coqc-prelude` .
